src/Microsoft.OData.Edm/Validation/EdmErrorCode.cs

```diff
@@ -1383,6 +1383,11 @@ public enum EdmErrorCode
         /// </summary>
         AnnotationApplyToNotAllowedAnnotatable = 400,
 
+        /// <summary>
+        /// A recursive complex-typed property must be optional.
+        /// </summary>
+        RecursiveComplexTypedPropertyMustBeOptional = 401,

```

Comment:
This will be a breaking change, you should probably move it to the end of the value list. You don't have to assign it a value, but if you do, make sure it is consistent with the other values at the end that aren't assigned values

Reply:
✅